plugins/modules/nd_flow_rules.py

```diff
@@ -0,0 +1,215 @@
+    flow_rule: my_FlowRule
+    subnets: []
+    state: present
+
+- name: Query a specific VRF Flow Rule
+  cisco.nd.nd_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    state: query
+
+- name: Query all VRF Flow Rules
+  cisco.nd.nd_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    state: query
+
+- name: Delete a VRF Flow Rule
+  cisco.nd.nd_flow_rules:
+    insights_group: my_ig
+    site_name: my_site
+    flow_rule: my_FlowRule
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        insights_group=dict(type="str", aliases=["fab_name", "ig_name"]),
+        site_name=dict(type="str", aliases=["site"]),
+        flow_rule=dict(type="str", aliases=["flow_rule_name", "name"]),  # Not required to query all objects
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        vrf=dict(type="str", aliases=["vrf_name"]),
+        subnets=dict(type="list", elements="str"),
+        state=dict(type="str", default="present", choices=["present", "absent", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+        required_if=[
+            ["state", "present", ["insights_group", "site_name", "flow_rule"]],
+            ["state", "absent", ["insights_group", "site_name", "flow_rule"]],
+            ["state", "query", ["insights_group", "site_name"]],
+        ],
+    )
+
+    nd = NDModule(module)
+    ndi = NDI(nd)
+
+    state = nd.params.get("state")
+    insights_group = nd.params.get("insights_group")
+    site_name = nd.params.get("site_name")
+    flow_rule = nd.params.get("flow_rule")
+    tenant = nd.params.get("tenant")
+    vrf = nd.params.get("vrf")
+    subnets = nd.params.get("subnets")
+
+    delete_keys = [
+        "links",
+        "uuid",
+        "flowRuleAttributeUuid",
+        "fabricName",
+    ]
+
+    trigger_path = ndi.config_ig_path + "/" + ndi.flow_rules_path.format(insights_group, site_name)
+    flow_rules_history = ndi.query_data(trigger_path)
+    nd.existing = {}
+    uuid = None
+    existing_subnets = []
+    for flow_rules_config in flow_rules_history:
+        if flow_rules_config.get("name") == flow_rule:
+            nd.existing = sanitize_dict(flow_rules_config, delete_keys)
+            uuid = flow_rules_config.get("uuid")
+            existing_subnets.extend(flow_rules_config.get("flowRuleAttributeList", []))
+
+    if state == "present":
+        nd.previous = nd.existing
+        flow_rule_config = {"name": flow_rule, "tenant": tenant, "vrf": vrf}
+        subnets_to_add, subnets_to_update = ndi.create_flow_rules_subnet_payload(subnets, existing_subnets)
```

Comment:
why do we need two return values in this function?

Reply:
I modified the behavior of this function. It returns only one list `subnets_to_update`.